src/nbl/ext/MitsubaLoader/CMitsubaMaterialCompilerFrontend.cpp

```diff
@@ -85,24 +86,24 @@ CMitsubaMaterialCompilerFrontend::EmitterNode* CMitsubaMaterialCompilerFrontend:
     if (_emitter->area.emissionProfile) {
         asset::material_compiler::IR::CEmitterNode::EmissionProfile profile;
         auto [image, sampler, meta] = getEmissionProfile(_emitter->area.emissionProfile);
-        if (!image) {
-            os::Printer::log("ERROR: Emission profile not loaded", ELL_ERROR);
-            return nullptr;
-        }
-        profile.texture = { image, sampler, 1.f };
-
-        auto inverseTransform = core::transpose(core::concatenateBFollowedByA(transform, _emitter->transform.matrix));
-        profile.right_hand = core::determinant(inverseTransform) >= 0.0f;
-        profile.up = core::normalize(inverseTransform[1]);
-        profile.view = core::normalize(inverseTransform[2]);
-
-        res->emissionProfile = profile;
-        float normalizeEnergy = _emitter->area.emissionProfile->normalizeEnergy;
-        if (normalizeEnergy < 0.0f) {
-            res->intensity *= meta->getMaxIntensity();
+        if (image) {
+            profile.texture = { image, sampler, 1.f };
+            auto inverseTransform = core::transpose(core::concatenateBFollowedByA(transform, _emitter->transform.matrix));
+            profile.right_hand = core::determinant(inverseTransform) >= 0.0f;
```

Comment:
we need to also check that the determinant is far away enough from 0.f (and issue an error message if it isn't) before deciding the profile is valid

https://github.com/Devsh-Graphics-Programming/Nabla/pull/630#discussion_r1456682313

Also compute the determinant of the matrix you're about to inverse, because if the inverse can't be found you'll get some garbage. This won't change the sign you're looking for as signs of reciprocals remain the same

Reply:
[resolved](https://github.com/Devsh-Graphics-Programming/Nabla/commit/b1ac324212a4b8dc3bf98a4797c58266f1ffc19a)